tests/drift/test_drift.py

```diff
@@ -841,3 +857,76 @@ def test_result_comparison_to_cbpe_plots_raise_no_exceptions(sample_drift_data):
         _ = result.compare(result2).plot()
     except Exception as exc:
         pytest.fail(f"an unexpected exception occurred: {exc}")
+
+
+def test_oo_behavior_univ_calc():  # noqa: D103
+    reference, monitored, _ = load_synthetic_car_loan_dataset()
+    reference2 = reference.copy(deep=True)
+    monitored2 = monitored.copy(deep=True)
+    feature_column_names = [
+        'car_value',
+        'salary_range',
+        'debt_to_income_ratio',
+        'loan_length',
+        'repaid_loan_on_prev_car',
+        'size_of_downpayment',
+        'driver_tenure'
+    ]
+    calc = UnivariateDriftCalculator(
+        column_names=feature_column_names,
+        treat_as_categorical=['y_pred'],
+        timestamp_column_name='timestamp',
+        continuous_methods=['kolmogorov_smirnov', 'jensen_shannon', 'wasserstein', 'hellinger'],
+        categorical_methods=['chi2', 'jensen_shannon', 'l_infinity', 'hellinger'],
+    )
+    calc.fit(reference2)
+    results = calc.calculate(monitored2)  # noqa: F841
+    pd.testing.assert_frame_equal(monitored, monitored2)
+    pd.testing.assert_frame_equal(reference, reference2)
+
+
+def test_oo_behavior_dre_calc():  # noqa: D103
+    reference, monitored, _ = load_synthetic_car_loan_dataset()
+    reference2 = reference.copy(deep=True)
+    monitored2 = monitored.copy(deep=True)
+    feature_column_names = [
+        'car_value',
+        'salary_range',
+        'debt_to_income_ratio',
+        'loan_length',
+        'repaid_loan_on_prev_car',
+        'size_of_downpayment',
+        'driver_tenure'
+    ]
+    calc = DataReconstructionDriftCalculator(
+        column_names=feature_column_names,
+        timestamp_column_name='timestamp',
+    )
+    calc.fit(reference2)
+    results = calc.calculate(monitored2)  # noqa: F841
+    pd.testing.assert_frame_equal(monitored, monitored2)
+    pd.testing.assert_frame_equal(reference, reference2)
+
+
+def test_oo_behavior_dc_calc():  # noqa: D103
```

Comment:
Name could be a bit more descriptive, as this is a weird thing to test.

Maybe something like test_input_dataframes_are_not_altered_by_calculator